tests/test_ecs/test_ecs_boto3.py

```diff
@@ -1757,6 +1764,111 @@ def test_list_task_definition_families():
     len(resp2['families']).should.equal(1)
 
 
+@mock_ec2
+@mock_ecs
+def test_default_container_instance_attributes():
+    ecs_client = boto3.client('ecs', region_name='us-east-1')
+    ec2 = boto3.resource('ec2', region_name='us-east-1')
+
+    test_cluster_name = 'test_ecs_cluster'
+
+    # Create cluster and EC2 instance
+    _ = ecs_client.create_cluster(
+        clusterName=test_cluster_name
+    )
+
+    test_instance = ec2.create_instances(
+        ImageId="ami-1234abcd",
+        MinCount=1,
+        MaxCount=1,
+    )[0]
+
+    instance_id_document = json.dumps(
+        ec2_utils.generate_instance_identity_document(test_instance)
+    )
+
+    # Register container instance
+    response = ecs_client.register_container_instance(
+        cluster=test_cluster_name,
+        instanceIdentityDocument=instance_id_document
+    )
+
+    response['containerInstance'][
+        'ec2InstanceId'].should.equal(test_instance.id)
+    full_arn = response['containerInstance']['containerInstanceArn']
+    container_instance_id = full_arn.rsplit('/', 1)[-1]
+
+    default_attributes = response['containerInstance']['attributes']
+    assert len(default_attributes) == 4
+    assert sorted(default_attributes) == sorted([
```

Comment:
Cant sort dictionaries like this. you need to add `key=lambda item: item['name']` as an argument to the sorted function so it knows what to sort on, do it for both calls to sorted.

Reply:
I can do that but this is a list of dictionaries. Even if it isn't sorting the way I expect, is that causing the errors I mentioned?  I'm having trouble understanding the connection between how that assertion error would fail and cause the Exception/AttributeError described above.

Reply:
The exception caused above appears because moto in server mode repurposes its Flask internals for pretending to be an API gateway, and a test is running a PATCH request, which is not supported, therefore flask returns a 500like error code and prints a stack trace, can be ignored.

Skip to line 9281 in the build output.

Reply:
I actually missed that, thanks for pointing it out and thanks for the review!